lxd/daemon.go

```diff
@@ -768,6 +771,9 @@ func (d *Daemon) init() error {
 		return errors.Wrap(err, "failed to open cluster database")
 	}
 
+	// TODO: Issue #6223: add logic for choosing xtables or nftables
+	d.firewall = iptables.XTables{}
```

Comment:
@keyallis if you added a factory function here using the go idoium of `NewX`, i.e `iptables.NewFirewall()` then you can will have a place to add the logic to choose xtables or nftables, you could then move the TODO comment there.

Reply:
More likely `firewall.New()` which would then return either a `xtables` or `nft` struct implementing the `Firewall` interface.

Reply:
This leads us into some dependency hell since xtables includes device and device includes firewall. One of the things we could do is move the xtables and nft struct definitions into the firewall package, but that doesn't seem like the most elegant solution to me. If that's what you think is best, or if you have another idea I can make that switch pretty quickly

Reply:
We wouldn't be able to move the struct definitions into firewall without moving the methods that implement Firewall, which rely on the device package. After a little work on this, the following is one of the import cycles we run into:
```
import cycle not allowed
package github.com/lxc/lxd/lxc-to-lxd
	imports github.com/lxc/lxd/lxd/migration
	imports github.com/lxc/lxd/lxd/operations
	imports github.com/lxc/lxd/lxd/state
	imports github.com/lxc/lxd/lxd/firewall
	imports github.com/lxc/lxd/lxd/iptables
	imports github.com/lxc/lxd/lxd/device
	imports github.com/lxc/lxd/lxd/storage
	imports github.com/lxc/lxd/lxd/migration
```

Reply:
@piluke so from what I can tell the only reason that these new packages need to import "lxd/device" is to access `device.NetworkSysctlGet()`. If that is the only reason then we could move those 2 functions (NetworkSysctlGet and NetworkSysctlSet) into a new package called "lxd/sysctl" potentially that would avoid the firewall or xtables packages importing the "lxd/device" package.

It may be worth making them more generic "sysctl.Set" and "sysctl.Get" and then removing the "/net" prefix inside the functions and adding it to where those functions are used today.

What do you think @stgraber ?

Reply:
Sounds good, though instead of a new package `lxd/util` would probably be fine to use for this purpose.

Reply:
I've done this for our relevant code. If I don't remove the function from `lxd/device/device_utils_network.go` then all existing code works exactly as it did before. I could leave it that way, or if you want I can switch all invocations to the `/lxd/util' package before committing and pushing up.

Reply:
@keyallis yes please update the invocations to the new package and remove the old functions.

Reply:
Just pushed those changes